test/index.test.js

```diff
@@ -327,14 +333,15 @@ const HasManyDissociator = require('../src/services/has-many-dissociator');
               filters: null,
             }, sequelizeOptions)
               .perform();
+            connectionManager.closeConnection();
```

Comment:
As test assertions are throwing errors, and as we can expect errors inside the tested code if something goes wrong, `connectionManager.closeConnection` is not ensured to be called.

If anything goes wrong, the connection will never be closed, and the test will never return (because you also removed the `forceExit` option)

If you want it to work in any case, you have 2 solutions:

1. Wrap all the test's code inside an try/finally block, and call `closeConnection` in the finally block
2. call `closeConnection` in a `afterEach` call, but this will not be allowed by jest linter.

Example for solution 1:

```js
      it('should generate a valid SQL query', async () => {
          expect.assertions(1);
          const { models, sequelizeOptions } = initializeSequelize();
          try{
            const stat = await new LineStatGetter(models.user, {
              type: 'Line',
              collection: 'user',
              filters: null,
            }, sequelizeOptions)
              .perform();
          } finally {
            connectionManager.closeConnection();
          }
          expect(stat.value).toHaveLength(1);
        });
```

Reply:
For solution 2:

```js
afterEach(() => {
  connectionManager.closeConnection()
})
```

I have a preference for solution 2 in this case, and I think it's a case where we want to bypass the linter's rule.

Reply:
According the discussions with @arnaudbesnier, @slimee and @rap2hpoutre, the decision is to not use jest hooks.